test/e2e/antreapolicy_test.go

```diff
@@ -65,12 +67,16 @@ func failOnError(err error, t *testing.T) {
 }
 
 // TestCase is a collection of TestSteps to be tested against.
+// If CleanupPerStep is set to false, policies created by each TestStep will not be cleaned up
+// right after that TestStep is completed. Instead, all types of policies and groups will be
+// cleaned up after the entire TestCase completes. This is useful when the TestSteps are additive.
 type TestCase struct {
-	Name  string
-	Steps []*TestStep
+	Name           string
+	Steps          []*TestStep
+	CleanupPerStep bool
```

Comment:
I'm typically of the opinion that if a resource should not be cleaned up by the test case, then it should be created outside of the test case itself. However, I understand that this may not play well with the framework or that it may require significant changes, so there is no need to address this here.

Reply:
The resources that's created by a testcase WILL always be cleaned up inside the testcase. The only difference here is whether it's cleaned up immediately after each `step`, or after all the `steps` in the testcase is executed. This makes testing pod/group update events easier in the framework.

Reply:
Yes I see the difference now between TestCase and TestStep now. So now it seems strange to me that we ever need to do automatic / generic cleanup at the end of a TestSteps... TestCases are supposed to be independent test units, why TestSteps should be incremental, building on the previous steps.

Reply:
It is a tricky problem. There are three scenarios here: 
1) `Testcase` and its `TestSteps` are all independent (most `Testcases` in this file). It really doesn't matter if policies are cleaned up after each step or at the end of the `Testcase`.
2) `Testcase` independent and `TestSteps` are additive. For this case I introduced the `CleanupPerStep=false`, so that all policies/groups are deleted at the end of the `Testcase`. Looping through the `TestSteps` and delete each policy will be troublesome because there will be policy overlapping there.
3) `Testcase` is dependent on underlying policies. For these `Testcases`, we do not want a hard cleanup on all the policies since the underlying policies will be deleted.

Now that I think about it more, maybe we can get around this problem by making sure that in scenario 2, subsequent deletion of a deleted resource in previous steps does not throw an error...